File: WebClients/EMS_Web/src/models/admintools/emailTemplateModel.ts
export class emailTemplateModel{
    Id!: number | 0;  
    FromEmail!:string|"";
    FromName!:string|"";
    TemplateCode!:string|"";
    TemplateSubject!:string|"";
    TemplateBody!:string|"";
    MailCCRoleIds!:string|"";
    IsActive!:boolean|false;
}
import { required, helpers } from '@vuelidate/validators'
class emailTemplateModelDTO {
    static model:emailTemplateModel=new emailTemplateModel();
    static rules = { 
        FromEmail: { required :helpers.withMessage('From email is required', required)},
        FromName: { required :helpers.withMessage('From name is required', required)},
        TemplateCode: { required :helpers.withMessage('Template Code is required', required)},
        TemplateSubject: { required :helpers.withMessage('Template Subject is required', required)},
        TemplateBody: { required :helpers.withMessage('Template Body is required', required)},
      }
}
export default emailTemplateModelDTO

File: Web Apps/TheTecniQ.Api/Models/Notification/EmailTemplateModel.cs
using FluentValidation;
using TheTecniQ.API.Models.Common;

namespace TheTecniQ.API.Models.Notification
{
    public class EmailTemplateModel : BaseModel
    {
        public string FromEmail { get; set; }
        public string FromName { get; set; }
        public string TemplateCode { get; set; }
        public string TemplateSubject { get; set; }
        public string TemplateBody { get; set; }
        public bool IsActive { get; set; }
    }
    public class EmailTemplateModelValidator : AbstractValidator<EmailTemplateModel>
    {
        public EmailTemplateModelValidator()
        {
            RuleFor(x => x.FromEmail).NotNull().NotEmpty().WithMessage("From email is required.");
            RuleFor(x => x.FromName).NotNull().NotEmpty().WithMessage("From name is required.");
            RuleFor(x => x.TemplateCode).NotNull().NotEmpty().WithMessage("Template code is required.");
            RuleFor(x => x.TemplateSubject).NotNull().NotEmpty().WithMessage("Template subject is required.");
            RuleFor(x => x.TemplateBody).NotNull().NotEmpty().WithMessage("Template body is required.");
        }
    }
}
